setup.py

```diff
@@ -175,7 +175,7 @@ def on_exit():
         define_macros=macros)
     extensions = [ext, posix_extension]
 # Solaris
-elif sys.platform.lower().startswith('sunos'):
+elif sys.platform.lower().startswith('sunos') or sys.platform.lower().startswith('solaris'):
```

Comment:
I believe you should do the same here: https://github.com/giampaolo/psutil/blob/master/test/test_psutil.py#L108